### PR TITLE
Give a hint of the language of the image to users

### LINKED ISSUE #3105: Give a hint of the language of the image to users (power users?)
When no ingredients are completed, powerusers and users have no way to see if the images are in the right language. Putting  the langcode in the alt description could be a way to solve this.
image

### PR DESCRIPTION
**Pull request by areeshatariq**
Signed-off-by: Areesha Tariq <areeshatariq02@gmail.com>

**Description:**
Added language code for the ingredient image that is being displayed.

**Related issues and discussion:** 
fixes #3105

### CODE REVIEW

File: lib/ProductOpener/Images.pm [hunk 1]
```diff
@@ -1314,7 +1314,7 @@ sub display_image($$$) {
 
 		my $path = product_path($product_ref);
 		my $rev = $product_ref->{images}{$id}{rev};
-		my $alt = remove_tags_and_quote($product_ref->{product_name}) . ' - ' . $Lang{$imagetype . '_alt'}{$lang};
+		my $alt = remove_tags_and_quote($product_ref->{product_name}) . ' - ' . $Lang{$imagetype . '_alt'}{$lang} . ' - ' . $display_lc;
```
Comment (stephanegigandet) [inline lib/ProductOpener/Images.pm]: So $display_lc is the language of the interface. e.g. on https://in.openfoodfacts.org, $display_lc is "en" (English). On https://fr.openfoodfacts.org , it is "fr".

But if we don't have an image in $display_lc, we will display an image for another language.

See this code:

# first try the requested language
	my @display_ids = ($imagetype . "_" . $display_lc);		my @display_ids = ($imagetype . "_" . $display_lc);
	# next try the main language of the product		# next try the main language of the product
	if ($product_ref->{lc} ne $display_lc) {		if ($product_ref->{lc} ne $display_lc) {
		push @display_ids, $imagetype . "_" . $product_ref->{lc};			push @display_ids, $imagetype . "_" . $product_ref->{lc};
	}		}

To solve this bug, we want to display in the alt tag the language of the actual image displayed, not the language of the interface.

Comment (areeshatariq) [reply lib/ProductOpener/Images.pm]: When I did not added image in $display_lc, while image was added for the other language, no ingredient image was displayed.

Comment (areeshatariq) [reply lib/ProductOpener/Images.pm]: @stephanegigandet When I did not add image in $display_lc, while images were added for other languages, no ingredient image was displayed.

Comment (areeshatariq) [reply lib/ProductOpener/Images.pm]: @stephanegigandet kindly review

File: lib/ProductOpener/Images.pm [hunk 2]
```diff
@@ -1315,6 +1315,12 @@ sub display_image($$$) {
 		my $path = product_path($product_ref);
 		my $rev = $product_ref->{images}{$id}{rev};
 		my $alt = remove_tags_and_quote($product_ref->{product_name}) . ' - ' . $Lang{$imagetype . '_alt'}{$lang};
+		if ($id eq ("ingredients_".$display_lc )) {
```
Comment (stephanegigandet) [inline lib/ProductOpener/Images.pm]:
```suggestion
		if ($id eq ($imagetype . "_" . $display_lc )) {
```
It's not only "ingrédients", it's also "front" and "nutrition" images.

Same for the other condition below.

Comment (areeshatariq) [reply lib/ProductOpener/Images.pm]: @stephanegigandet oh I thought we are doing it for ingredient image only. I've added for the other condition too so alt can include language code for front and nutrition images also.

**Pull request merged**